packages/datadog-plugin-jest/test/circus.spec.js

```diff
@@ -1,659 +1,306 @@
-              [TEST_TYPE]: 'test',
-              [ERROR_TYPE]: 'Error',
-              [ERROR_MESSAGE]: 'hook failure',
-              [ERROR_STACK]: hookError.stack
-            })
-          }).then(done).catch(done)
-
-        const hookFailureEvent = {
-          name: 'hook_failure',
-          test: {
-            fn: () => {},
-            name: TEST_NAME,
-            errors: [[
-              'hook failure',
-              hookError
-            ]]
-          }
-        }
-
-        datadogJestEnv.handleTestEvent(hookFailureEvent)
-      })
-
-      it('does not crash when there is an error in a hook outside a test', (done) => {
-        const hookFailureEvent = {
-          name: 'hook_failure'
-        }
-        datadogJestEnv.handleTestEvent(hookFailureEvent).then(done).catch(done)
-      })
-
-      it('should not crash when getVmContext is not a function', (done) => {
-        const testStartEvent = {
-          name: 'test_start',
-          test: {
-            fn: () => {},
-            name: TEST_NAME
-          }
-        }
-        datadogJestEnv.getVmContext = undefined
-
-        datadogJestEnv.handleTestEvent(testStartEvent)
-        testStartEvent.test.fn()
-
-        agent
-          .use(traces => {
-            expect(traces[0][0].meta[ERROR_TYPE]).to.be.undefined
-            expect(traces[0][0].meta[TEST_NAME_TAG]).to.equal(TEST_NAME)
-          }).then(done).catch(done)
-      })
-
-      // TODO: allow the plugin consumer to define their own jest's `testEnvironment`
-      it.skip('should allow the customer to use their own environment', (done) => {
-        class CustomerCustomEnv extends DatadogJestEnvironment {
-          async handleTestEvent () {
-            // do custom stuff
-            return new Promise(resolve => {
-              setTimeout(resolve, 100)
-            })
-          }
-        }
-        const customerTestEnv = new CustomerCustomEnv({ rootDir: BUILD_SOURCE_ROOT }, { testPath: TEST_SUITE })
-        agent
-          .use(traces => {
-            expect(traces[0][0].meta).to.contain({
-              language: 'javascript',
-              service: 'test',
-              [TEST_FRAMEWORK]: 'jest',
-              [TEST_NAME_TAG]: TEST_NAME,
-              [TEST_STATUS]: 'fail',
-              [TEST_SUITE_TAG]: TEST_SUITE,
-              [TEST_TYPE]: 'test'
-            })
-            expect(traces[0][0].type).to.equal('test')
-            expect(traces[0][0].name).to.equal('jest.test')
-            expect(traces[0][0].resource).to.equal(`${TEST_SUITE}.${TEST_NAME}`)
-          }).then(done).catch(done)
-        const testEvent = {
-          name: 'test_start',
-          test: {
-            fn: () => {},
-            name: TEST_NAME
-          }
-        }
-        customerTestEnv.handleTestEvent(testEvent)
-        testEvent.test.fn()
+        jestExecutable.runCLI(
+          options,
+          options.projects
+        )
       })
+      // TODO stack frames include our files??
```

Comment:
this file has changed a great deal. We used to manually call `datadogJestEnv.handleTestEvent` to test, which was quite bad (this is not the way our users use this plugin). We now programmatically run `jest` and see the side effects, the same way we do with `jasmine` tests.